Upgraded to pgx v5 and bump package major version to v3

niallnsec opened the pull request:
This pull requests updates the pgx dependency to the v5 branch as v4 is out of support. I have also bumped the package version to v3 as this is a breaking change.

hsluoyz commented:
@pckhoi plz review

pckhoi commented on adapter.go:
```diff
@@ -134,16 +134,22 @@ func (a *Adapter) LoadPolicy(model model.Model) error {
+	rows.Close()
+	if err == nil {
+		err = rows.Err()
+	}
```
Could have replaced with `ForEachRow`.

niallnsec replied:
Updated with latest commits

pckhoi commented on adapter.go:
```diff
@@ -201,19 +207,26 @@ func (a *Adapter) SavePolicy(model model.Model) error {
-			[]string{"id", "p_type", "v0", "v1", "v2", "v3", "v4", "v5"},
-			pgx.CopyFromRows(rows),
-		)
+	tx, err := a.pool.Begin(ctx)
+	if err != nil {
+		return err
+	}
+	_, err = tx.Exec(context.Background(), fmt.Sprintf("DELETE FROM %s WHERE id IS NOT NULL", a.schemaTable()))
+	if err != nil {
+		tx.Rollback(context.Background())
+		return err
+	}
+	_, err = tx.CopyFrom(
+		context.Background(),
+		a.tableIdentifier(),
+		[]string{"id", "p_type", "v0", "v1", "v2", "v3", "v4", "v5"},
+		pgx.CopyFromRows(rows),
+	)
+	if err != nil {
+		tx.Rollback(context.Background())
```
Should replace with `pgx.BeginFunc`.

niallnsec replied:
Updated with latest commits

niallnsec commented:
Sorry, my bad. I saw that the functions used previously had been removed and didn't see the top level funcs in the pgx package to replace them, which is why I reverted to the more explicit approach.

I've changed it so that it mirrors more appropriately your existing implementation.

pckhoi commented:
@hsluoyz how do you feel about bumping this adapter to v3 while still using casbin v2?

hsluoyz commented:
@pckhoi it's OK. Adapter version and Casbin version are not related. See: https://github.com/casbin/redis-adapter

pckhoi commented:
@niallnsec looks like the Go version is incompatible with pgx v5. Please upgrade go version in go.mod to 1.18 and up.

niallnsec commented:
@pckhoi I already did update the go.mod version to 1.18

niallnsec commented:
I noticed the CI was failing because I hadn't bumped the Go version from 1.14 to 1.18, so I have now done that so hopefully the tests will run
